tests/strings/test_z_algorithm.py

```diff
@@ -1,21 +1,18 @@
-from random import choice
 from itertools import product
+from random import choice, randint
 from string import ascii_lowercase
 
 from pyrival.strings.z_algorithm import z_function
 
 
-def brute_force(text, limit=2**30):
-    k = min(limit, len(text))
-    z_array = [0] * k
+def get_lcp(text, i):
```

Comment:
Just have something like 
```
def Z_brute(S):
    n = len(S)
    
    Z = [0]
    for i in range(1, n):
        z = 0
        while i + z < n and S[i] == S[i + z]:
            z += 1
        Z.append(z)
    return Z
```

Reply:
It's the same thing, no? Except I only moved the inner loop inside a function, which I'm also re-using in the large test. It's only an utility function to assert correctness, so not doing this suggested change.

Reply:
In general I like the idea of having a brute O(n^2) version that behaves exactly like the fast O(n) is supposed to do. But do as you wish here, it doesn't really matter.